CompressionEngine.java

```diff
@@ -0,0 +1,483 @@
+// CompressionEngine.java
+// Author: Eric Zhuang
+// CodeU Project Group 6
+
+//Converts all aspects of a message into a string format
+//Compresses the message content into a more storage-efficient format
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+
+import codeu.chat.common.Time;
+import codeu.chat.common.Uuid;
+import codeu.chat.common.Uuids;
+
+//TODO: Figure out where to put this in the pipeline
+// Message.java read/write
+// Server.java sendToRelay
+
+public class CompressionEngine{
+
+	public static String[] compressMessage(Message msg){}
+	public static Message decompressMessage(String[] packet){}
+
+	private String compressString(String message){
+		return LZString.compress(message);
+	}
+
+	private String convertUuid(Uuid id){
+		return id.toString();
+	}
+
+	private String convertTime (Time time){
+		return time.toString();
+	}
+
+	private String decompressString(String message){
+		return LZString.decompress(String(message));
+	}
+
+	private Uuid convertToUuid(String id){
+		return Uuid.fromString(id);
+	}
+
+	private Time convertToTime(String time){
+		return Time.fromString(time); // Will need to create this function
+	}
+}
+
+// Class for decompressing strings, taken from https://github.com/diogoduailibe/lzstring4j
+public class LZString {
+
+
+  static String keyStr = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
+
+
+  public static String compress(String uncompressed) {
```

Comment:
I'm sorry but this is a really long method. Please break it down into smaller methods, with comments indicating what each of those methods do.